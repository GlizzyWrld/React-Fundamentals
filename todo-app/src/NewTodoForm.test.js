import { render, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

test('renders NewTodoForm component without errors', () => {
    const { container } = render(<NewTodoForm addTodo={() => {}} />);
    expect(container).toMatchSnapshot();
  });
  
  test('submits form and calls addTodo function with correct value', () => {
    const addTodoMock = jest.fn();
    const { container, getByPlaceholderText, getByText } = render(
      <NewTodoForm addTodo={addTodoMock} />
    );
    const taskInput = getByPlaceholderText('Enter task');
    const addButton = getByText('Add Todo');
  
    fireEvent.change(taskInput, { target: { value: 'Test Task' } });
    fireEvent.click(addButton);
  
    expect(addTodoMock).toHaveBeenCalledWith('Test Task');
    expect(container).toMatchSnapshot();
  });
  
