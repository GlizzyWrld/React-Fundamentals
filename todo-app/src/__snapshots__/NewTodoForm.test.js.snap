// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NewTodoForm component without errors 1`] = `
<div>
  <form>
    <input
      placeholder="Enter task"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</div>
`;

exports[`submits form and calls addTodo function with correct value 1`] = `
<div>
  <form>
    <input
      placeholder="Enter task"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</div>
`;
